app/api/items/create_item.go

```diff
@@ -521,19 +530,26 @@ func registerChildrenValidator(formData *formdata.FormData, store *database.Data
 }
 
 func (srv *Service) insertItem(store *database.DataStore, user *database.User, formData *formdata.FormData,
-	newItemRequest *NewItemRequest) (itemID int64) {
+	newItemRequest *NewItemRequest) (itemID int64, apiError service.APIError) {
 	itemMap := formData.ConstructPartialMapForDB("ItemWithRequiredType")
 	stringMap := formData.ConstructPartialMapForDB("newItemString")
 
-	service.MustNotBeError(store.WithForeignKeyChecksDisabled(func(fkStore *database.DataStore) error {
+	err := store.WithForeignKeyChecksDisabled(func(fkStore *database.DataStore) error {
 		return fkStore.RetryOnDuplicatePrimaryKeyError(func(s *database.DataStore) error {
 			itemID = s.NewID()
 
 			itemMap["id"] = itemID
 			itemMap["default_language_tag"] = newItemRequest.LanguageTag
 			return s.Items().InsertMap(itemMap)
 		})
-	}))
+	})
+	e, ok := err.(*mysql.MySQLError)
+	if ok && e.Number == 1062 {
```

Comment:
Looks like a magic number, shouldn't it be a reusable constant in a dedicated file?

Reply:
fixed everywhere

Reply:
👍